src/shogun/lib/StoppableSGObject.h

```diff
@@ -32,14 +32,16 @@ namespace shogun
 		/** destructor */
 		virtual ~CStoppableSGObject();
 
-		/** init flags to false as default */
-		void init_stoppable();
-
 #ifndef SWIG
 		/** @return whether the algorithm needs to be stopped */
 		SG_FORCED_INLINE bool cancel_computation() const
 		{
-			return m_cancel_computation.load();
+			/* Execute the callback, if present*/
+			bool result_callback = false;
+			if (m_callback)
+				result_callback = m_callback();
+
+			return m_cancel_computation.load() || result_callback;
```

Comment:
since it's `||` maybe it'd be better to have this as:
```
return (m_callback) ? (m_cancel_computation.load() || result_callback()) : m_cancel_computation.load();
```

just for the sake of lazy evaluation of or :)